flink-streaming-java/src/main/java/org/apache/flink/streaming/runtime/translators/SinkTransformationTranslator.java

```diff
@@ -0,0 +1,283 @@
+			Context context) {
+
+		if (!sinkTransformation.getSink().getGlobalCommittableSerializer().isPresent()) {
+			return;
+		}
+
+		final TypeInformation<CommT> committableTypeInfo = TypeExtractor.createTypeInfo(
+				Sink.class,
+				sinkTransformation.getSink().getClass(),
+				1,
+				null,
+				null);
+
+		addOperatorToStreamGraph(
+				globalCommitterFactory, inputId,
+				committableTypeInfo,
+				null,
+				"Sink Global Committer:",
+				1,
+				1,
+				sinkTransformation,
+				context);
+	}
+
+	private int getParallelism(
+			SinkTransformation<InputT, CommT, WriterStateT, GlobalCommT> sinkTransformation,
+			Context context) {
+		return sinkTransformation.getParallelism() != ExecutionConfig.PARALLELISM_DEFAULT
+				? sinkTransformation.getParallelism()
+				: context.getStreamGraph().getExecutionConfig().getParallelism();
+	}
+
+	/**
+	 * Add a operator to the {@link StreamGraph}.
+	 *
+	 * @param operatorFactory The operator factory
+	 * @param inputId The upstream stream node id of the operator
+	 * @param inTypeInfo The input type information of the operator
+	 * @param outTypInfo The output type information of the operator
+	 * @param prefix The prefix of the name and uid of the operator
+	 * @param parallelism The parallelism of the operator
+	 * @param maxParallelism The max parallelism of the operator
+	 * @param sinkTransformation The sink transformation which the operator belongs to
+	 *
+	 * @return The stream node id of the operator
+	 */
+	private <IN, OUT> int addOperatorToStreamGraph(
+			StreamOperatorFactory<OUT> operatorFactory, int inputId,
+			TypeInformation<IN> inTypeInfo,
+			TypeInformation<OUT> outTypInfo,
+			String prefix,
+			int parallelism,
+			int maxParallelism,
+			SinkTransformation<InputT, CommT, WriterStateT, GlobalCommT> sinkTransformation,
+			Context context) {
+		final StreamGraph streamGraph = context.getStreamGraph();
+		final String slotSharingGroup = context.getSlotSharingGroup();
+		final int transformationId = Transformation.getNewNodeId();
+
+		final ChainingStrategy chainingStrategy = sinkTransformation.getChainingStrategy();
+		if (chainingStrategy != null) {
+			operatorFactory.setChainingStrategy(chainingStrategy);
+		}
+
+		streamGraph.addOperator(
+				transformationId,
+				slotSharingGroup,
+				sinkTransformation.getCoLocationGroupKey(),
+				operatorFactory,
+				inTypeInfo,
+				outTypInfo,
+				String.format("%s %s", prefix, sinkTransformation.getName()));
+
+		streamGraph.setParallelism(transformationId, parallelism);
+		streamGraph.setMaxParallelism(transformationId, maxParallelism);
+
+		StreamGraphUtils.configureBufferTimeout(
+				streamGraph,
+				transformationId,
+				sinkTransformation,
+				context.getDefaultBufferTimeout());
+		if (sinkTransformation.getUid() != null) {
+			streamGraph.setTransformationUID(
+					transformationId,
+					String.format("%s %s", prefix, sinkTransformation.getUid()));
+		}
```

Comment:
Do we know what happens if nothing is set?

Reply:
IIRC, we could not reuse the state if we not set the uid.

Reply:
I see. I was wondering then if we should force the user to set it, but this would be different behaviour than the rest of the transformations, so we may not want to do it.